`state mv` on TF 0.15.4 causes later plans to crash with `value is not known`

### Terraform Version

```
Terraform v0.15.4
on darwin_amd64
```

### Terraform Configuration Files

(Sorry, my terraform configuration pulls in lots of internal modules from multiple private repositories and is very hard to trim down to something sharable).

### Debug Output

```
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_read_access module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_read_access["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].aws_iam_policy.elasticsearch_read_access" to "module.elasticgraph_clusters[\"payplat\"].aws_iam_policy.elasticsearch_read_access[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_write_access module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_write_access["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].aws_iam_policy.elasticsearch_write_access" to "module.elasticgraph_clusters[\"payplat\"].aws_iam_policy.elasticsearch_write_access[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].random_shuffle.subnet_ids module.elasticgraph_clusters["payplat"].random_shuffle.subnet_ids["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].random_shuffle.subnet_ids" to "module.elasticgraph_clusters[\"payplat\"].random_shuffle.subnet_ids[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.elasticsearch.aws_elasticsearch_domain.es module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].aws_elasticsearch_domain.es' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch.aws_elasticsearch_domain.es" to "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch[\"payments1\"].aws_elasticsearch_domain.es"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.elasticsearch.aws_security_group.es module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].aws_security_group.es' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch.aws_security_group.es" to "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch[\"payments1\"].aws_security_group.es"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.elasticsearch.null_resource.register_to_sqs[0] module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].null_resource.register_to_sqs[0]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch.null_resource.register_to_sqs[0]" to "module.elasticgraph_clusters[\"payplat\"].module.elasticsearch[\"payments1\"].null_resource.register_to_sqs[0]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"] module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"read\"]" to "module.elasticgraph_clusters[\"payplat\"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"] module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"write\"]" to "module.elasticgraph_clusters[\"payplat\"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"] module.elasticgraph_clusters["payplat"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"read\"]" to "module.elasticgraph_clusters[\"payplat\"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"] module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"read\"]" to "module.elasticgraph_clusters[\"payplat\"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"] module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"write\"]" to "module.elasticgraph_clusters[\"payplat\"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"] module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"read\"]" to "module.elasticgraph_clusters[\"payplat\"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform state mv -lock=true module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"] module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'
Move "module.elasticgraph_clusters[\"payplat\"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch[\"write\"]" to "module.elasticgraph_clusters[\"payplat\"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch[\"payments1\"]"
Successfully moved 1 object(s).
Running command '/usr/local/bin/terraform plan -var-file=../../tfvars/development/development.tfvars.json -var-file=../../tfvars/development/development-us-west-2.tfvars.json' in directory '/Users/myron/Development/tf-elasticgraph/./components/main'

[refresh output redacted]

panic: value is not known
goroutine 4087 [running]:
github.com/zclconf/go-cty/cty.Value.LengthInt(0x38b7c48, 0xc001272790, 0x3031c40, 0x49a9b60, 0x0)
        /go/pkg/mod/github.com/zclconf/go-cty@v1.8.3/cty/value_ops.go:1060 +0x2a7
github.com/hashicorp/terraform/internal/plans/objchange.assertPlanValid(0xc001912780, 0x38b7cb8, 0xc001273d10, 0x305ca60, 0xc002f3d6b0, 0x38b7cb8, 0xc001273670, 0x305ca60, 0xc002f3cc90, 0x38b7cb8, ...)
        /home/circleci/project/project/internal/plans/objchange/plan_valid.go:92 +0xb15
github.com/hashicorp/terraform/internal/plans/objchange.assertPlanValid(0xc001907800, 0x38b7cb8, 0xc0013de280, 0x305ca60, 0xc002f3dc80, 0x38b7cb8, 0xc0012736b0, 0x305ca60, 0xc002f3cd50, 0x38b7cb8, ...)
        /home/circleci/project/project/internal/plans/objchange/plan_valid.go:113 +0x11af
github.com/hashicorp/terraform/internal/plans/objchange.AssertPlanValid(...)
        /home/circleci/project/project/internal/plans/objchange/plan_valid.go:36
github.com/hashicorp/terraform/internal/terraform.(*NodeAbstractResourceInstance).plan(0xc0033c0000, 0x38eee80, 0xc0017c08c0, 0x0, 0xc000ef9500, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
        /home/circleci/project/project/internal/terraform/node_resource_abstract_instance.go:778 +0x17bd
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).managedResourceExecute(0xc0031186c0, 0x38eee80, 0xc0017c08c0, 0x49, 0xc00398e780, 0xc003c4fc50)
        /home/circleci/project/project/internal/terraform/node_resource_plan_instance.go:196 +0x8cb
github.com/hashicorp/terraform/internal/terraform.(*NodePlannableResourceInstance).Execute(0xc0031186c0, 0x38eee80, 0xc0017c08c0, 0xc0001b0002, 0xc003c4fce8, 0x100b805, 0x30a8180)
        /home/circleci/project/project/internal/terraform/node_resource_plan_instance.go:54 +0xb3
github.com/hashicorp/terraform/internal/terraform.(*ContextGraphWalker).Execute(0xc0022e8680, 0x38eee80, 0xc0017c08c0, 0x5abffc8, 0xc0031186c0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/internal/terraform/graph_walk_context.go:129 +0xbf
github.com/hashicorp/terraform/internal/terraform.(*Graph).walk.func1(0x33fbbc0, 0xc0031186c0, 0x0, 0x0, 0x0)
        /home/circleci/project/project/internal/terraform/graph.go:59 +0xbd3
github.com/hashicorp/terraform/internal/dag.(*Walker).walkVertex(0xc000fa73e0, 0x33fbbc0, 0xc0031186c0, 0xc0047ab040)
        /home/circleci/project/project/internal/dag/walk.go:381 +0x288
created by github.com/hashicorp/terraform/internal/dag.(*Walker).Update
        /home/circleci/project/project/internal/dag/walk.go:304 +0x1246
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.920246278.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.
When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.
SECURITY WARNING: the "crash.920246278.log" file that was created may contain
sensitive information that must be redacted before it is safe to share
on the issue tracker.
[1]: https://github.com/hashicorp/terraform/issues
!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

### Crash Output

The crash log is 70K lines and contains many things I'd need to redact before I can share it (which would take me a long time, given its length). I'm hoping the stacktrace from the panic is sufficient for diagnosing this issue.

### Expected Behavior

After the `state mv` operations are applied, a `terraform plan` should not crash.

### Actual Behavior

After applying a series of `state mv` operations, `terraform plan` crashes with the panic shown above. And worst of all, it crashes for every attempt at running a `plan` after that.  Blowing away the `.terraform` directory doesn't help.  The only way that we've found to solve it is to manually restore our state file in S3 to its prior version (luckily, we are using a versioned S3 bucket).   Once that is done `plan` works again--but it also undoes the `state mv` operations I need to perform.

### Steps to Reproduce

Unfortunately, I'm not able to get this trimmed down to a series of steps I can provide to reproduce the issue given how large my terraform project is (and also how slow it is to iterate: I don't actually have access to the S3 bucket to restore my state file to be able to try anything different, and I have to ask another engineer at my organization to restore it after any attempts).  Still, here's roughly the steps I went through:

1. Refactored my terraform code to introduce the use of `foreach` when calling a module that provisions and `aws_elasticsearch_domain` (and also in some related `resource` blocks).
2. Ran a successful plan which told me that 13 resources would be destroyed and recreated (due to the resource paths on them changing).
3. Wrote a script that applies a series of `state mv` commands to move each of the 13 resources from its old path to its new path.  The script also includes a `plan` at the end.
4. Ran the script. It appeared to successfully perform each `state mv` operation, but while executing the `plan` at the end, terraform crash with the output shown above.
5. Tried to run a `plan` manually; again it crashed with similar output to what is shown above. This happens every time I run a `plan`--like my state file is corrupted somehow or something.

Here's the terraform commands in my script:

```
state mv -lock=true 'module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_read_access' 'module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_read_access["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_write_access' 'module.elasticgraph_clusters["payplat"].aws_iam_policy.elasticsearch_write_access["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].random_shuffle.subnet_ids' 'module.elasticgraph_clusters["payplat"].random_shuffle.subnet_ids["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.elasticsearch.aws_elasticsearch_domain.es' 'module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].aws_elasticsearch_domain.es'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.elasticsearch.aws_security_group.es' 'module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].aws_security_group.es'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.elasticsearch.null_resource.register_to_sqs[0]' 'module.elasticgraph_clusters["payplat"].module.elasticsearch["payments1"].null_resource.register_to_sqs[0]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"]' 'module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"]' 'module.elasticgraph_clusters["payplat"].module.execute_rake_task_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"]' 'module.elasticgraph_clusters["payplat"].module.graphql_endpoint_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"]' 'module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"]' 'module.elasticgraph_clusters["payplat"].module.indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["read"]' 'module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_read_access_to_elasticsearch["payments1"]'
state mv -lock=true 'module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_access_to_elasticsearch["write"]' 'module.elasticgraph_clusters["payplat"].module.kafka_proto_indexer_lambda.aws_iam_role_policy_attachment.grant_write_access_to_elasticsearch["payments1"]'
plan
```

### Additional Context


### References

--- a/internal/plans/objchange/plan_valid.go
+++ b/internal/plans/objchange/plan_valid.go
@@ -69,7 +69,20 @@ func assertPlanValid(schema *configschema.Block, priorState, config, plannedStat
 			// Easy path: nothing has changed at all
 			continue
 		}
+
+		if !configV.IsKnown() {
+			// An unknown config block represents a dynamic block where the
+			// for_each value is unknown, and therefor cannot be altered by the
+			// provider.
+			errs = append(errs, path.NewErrorf("planned value %#v for unknown dynamic block", plannedV))
+			continue
+		}
+
 		if !plannedV.IsKnown() {
+			// Only dynamic configuration can set blocks to unknown, so this is
+			// not allowed from the provider. This means that either the config
+			// and plan should match, or we have an error where the plan
+			// changed the config value, both of which have been checked.
 			errs = append(errs, path.NewErrorf("attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead"))
 			continue
 		}
@@ -94,6 +107,7 @@ func assertPlanValid(schema *configschema.Block, priorState, config, plannedStat
 				errs = append(errs, path.NewErrorf("block count in plan (%d) disagrees with count in config (%d)", plannedL, configL))
 				continue
 			}
+
 			for it := plannedV.ElementIterator(); it.Next(); {
 				idx, plannedEV := it.Element()
 				path := append(path, cty.IndexStep{Key: idx})

--- a/internal/plans/objchange/plan_valid_test.go
+++ b/internal/plans/objchange/plan_valid_test.go
@@ -388,6 +388,110 @@ func TestAssertPlanValid(t *testing.T) {
 				`.b: attribute representing a list of nested blocks must be empty to indicate no blocks, not null`,
 			},
 		},
+
+		// blocks can be unknown when using dynamic
+		"nested list, unknown nested dynamic": {
+			&configschema.Block{
+				BlockTypes: map[string]*configschema.NestedBlock{
+					"a": {
+						Nesting: configschema.NestingList,
+						Block: configschema.Block{
+							BlockTypes: map[string]*configschema.NestedBlock{
+								"b": {
+									Nesting: configschema.NestingList,
+									Block: configschema.Block{
+										Attributes: map[string]*configschema.Attribute{
+											"c": {
+												Type:     cty.String,
+												Optional: true,
+											},
+											"computed": {
+												Type:     cty.String,
+												Computed: true,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+					"computed": cty.NullVal(cty.String),
+					"b": cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+						"c": cty.StringVal("x"),
+					})}),
+				})}),
+			}),
+			cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+					"b": cty.UnknownVal(cty.List(cty.Object(map[string]cty.Type{
+						"c":        cty.String,
+						"computed": cty.String,
+					}))),
+				})}),
+			}),
+			cty.ObjectVal(map[string]cty.Value{
+				"a": cty.ListVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+					"b": cty.UnknownVal(cty.List(cty.Object(map[string]cty.Type{
+						"c":        cty.String,
+						"computed": cty.String,
+					}))),
+				})}),
+			}),
+			[]string{},
+		},
+
+		"nested set, unknown dynamic cannot be planned": {
+			&configschema.Block{
+				Attributes: map[string]*configschema.Attribute{
+					"computed": {
+						Type:     cty.String,
+						Computed: true,
+					},
+				},
+				BlockTypes: map[string]*configschema.NestedBlock{
+					"b": {
+						Nesting: configschema.NestingSet,
+						Block: configschema.Block{
+							Attributes: map[string]*configschema.Attribute{
+								"c": {
+									Type:     cty.String,
+									Optional: true,
+								},
+							},
+						},
+					},
+				},
+			},
+
+			cty.ObjectVal(map[string]cty.Value{
+				"computed": cty.NullVal(cty.String),
+				"b": cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+					"c": cty.StringVal("x"),
+				})}),
+			}),
+			cty.ObjectVal(map[string]cty.Value{
+				"computed": cty.NullVal(cty.String),
+				"b": cty.UnknownVal(cty.Set(cty.Object(map[string]cty.Type{
+					"c": cty.String,
+				}))),
+			}),
+			cty.ObjectVal(map[string]cty.Value{
+				"computed": cty.StringVal("default"),
+				"b": cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{
+					"c": cty.StringVal("oops"),
+				})}),
+			}),
+
+			[]string{
+				`.b: planned value cty.SetVal([]cty.Value{cty.ObjectVal(map[string]cty.Value{"c":cty.StringVal("oops")})}) for unknown dynamic block`,
+			},
+		},
+
 		"nested set, null in plan": {
 			&configschema.Block{
 				BlockTypes: map[string]*configschema.NestedBlock{

